src/wrappers/dashboards.ts

```diff
@@ -163,27 +234,171 @@ export default class {
+
+    const includedResources = content.included || [];
+
+    const labelsToCreate = includedResources.reduce((acc, ir) => {
+      if (ir.type === TemplateType.Label) {
+        const found = labelRelationships.some((lr) => lr.type === TemplateType.Label && lr.id === ir.id);
+        if (found) {
+          acc = [...acc, ir];
+        }
+      }
+      return acc;
+    }, [] as ILabelIncluded[]);
+
+    const pendingLabels = labelsToCreate.map((l) => {
+      const { attributes: { name, properties } } = l;
+      return this.labelsService.labelsPost({ name, properties });
+    });
+
+    const labelsResponse = await Promise.all(pendingLabels);
+
+    const createdLabels = labelsResponse
+      .map((lr) => lr.data.label)
+      .filter((cl): cl is Label => !!cl)
+      .map((l) => l.id)
+      .filter((id): id is string => !!id);
+
+    if (!dashboard || !dashboard.id) {
+      throw new Error("Can not add labels to undefined Dashboard");
+    }
+
+    await this.addLabels(dashboard.id, createdLabels);
+  }
+
+  private async createIncludedCellsFromTemplate(template: IDashboardTemplate,
+                                                createdDashboard: IDashboard) {
+
+    const { content } = template;
+
+    if (
+      !content.data.relationships ||
+      !content.data.relationships[TemplateType.Cell]
+    ) {
+      return;
+    }
+
+    const cellRelationships =
+      content.data.relationships[TemplateType.Cell].data;
+
+    const includedResources = content.included || [];
+
+    const cellsToCreate = includedResources.reduce((acc, ir) => {
+      if (ir.type === TemplateType.Cell) {
+        const found = cellRelationships.some((cr) => cr.type === TemplateType.Cell && cr.id === ir.id);
+        if (found) {
+          acc = [...acc, ir];
+        }
+      }
+      return acc;
+    }, [] as ICellIncluded[]);
+
+    const pendingCells = cellsToCreate.map((c) => {
+      const { attributes: { name, x, y, w, h } } = c;
+      return this.createCell(createdDashboard.id, { name, x, y, w, h });
+    });
+
+    const cellResponses = await Promise.all(pendingCells);
+    this.createIncludedViewsFromTemplate(template, cellResponses, cellsToCreate, createdDashboard);
+  }
+
+  private async createIncludedViewsFromTemplate(
+    template: IDashboardTemplate,
+    createdCells: Cell[],
+    originalCellsIncluded: ICellIncluded[],
+    createdDashboard: IDashboard,
+  ) {
+    const pendingViews = createdCells.map((c, i) => {
+      const cellFromTemplate = originalCellsIncluded[i];
+      const viewRelationship = cellFromTemplate.relationships[TemplateType.View].data;
+      const includedResources = template.content.included || [];
+
+      const includedView = includedResources.find((ir) => {
+        return (ir.type === TemplateType.View && ir.id === viewRelationship.id);
+      });
+
+      if (includedView) {
+        return this.updateView(createdDashboard.id, c.id || "", includedView.attributes);
```

Comment:
Will this create the view if it doesn't already exist? `updateView` seems like it will update a view that already exists.

Reply:
yes, creating a cell creates a view.